src/main/java/net/neoforged/neoforge/transfer/handlers/wrappers/items/WorldlyContainerWrapper.java

```diff
@@ -0,0 +1,120 @@
+            throw new IllegalArgumentException("Cannot access worldly container on side " + side + " : out of bounds slot index " + slot + " with size " + slots.length);
+        }
+        return slots[slot];
+    }
+
+    @Override
+    public int size() {
+        return side == null ? worldlyContainer.getContainerSize() : worldlyContainer.getSlotsForFace(side).length;
+    }
+
+    @Override
+    public ItemResource getResource(int index) {
+        return wrappedContainer.getResource(convertSlot(index));
+    }
+
+    @Override
+    public int getAmount(int index) {
+        return wrappedContainer.getAmount(convertSlot(index));
+    }
+
+    @Override
+    public int getCapacity(int index, ItemResource resource) {
+        return wrappedContainer.getCapacity(convertSlot(index), resource);
+    }
+
+    @Override
+    public boolean isValid(int index, ItemResource resource) {
+        return wrappedContainer.isValid(convertSlot(index), resource);
+    }
+
+    @Override
+    public boolean supportsInsertion(int index) {
+        return wrappedContainer.supportsInsertion(convertSlot(index));
+    }
+
+    @Override
+    public boolean supportsExtraction(int index) {
+        return wrappedContainer.supportsExtraction(convertSlot(index));
+    }
+
+    @Override
+    public int insert(int index, ItemResource resource, int amount, TransactionContext transaction) {
+        int convertedIndex = convertSlot(index);
+        if (!worldlyContainer.canPlaceItemThroughFace(convertedIndex, UnsafeResourceUtils.innerStackOf(resource), side)) {
+            return 0;
+        }
+        return wrappedContainer.insert(convertedIndex, resource, amount, transaction);
+    }
+
+    @Override
+    public int insert(ItemResource resource, int amount, TransactionContext transaction) {
+        int size = size();
+        int handled = 0;
+        for (int index = 0; index < size; index++) {
+            handled += insert(index, resource, amount - handled, transaction);
+            if (handled == amount) break;
+        }
+        return handled;
+    }
+
+    @Override
+    public int extract(int index, ItemResource resource, int amount, TransactionContext transaction) {
+        int convertedSlot = convertSlot(index);
+        if (side != null && !worldlyContainer.canTakeItemThroughFace(convertedSlot, UnsafeResourceUtils.innerStackOf(resource), side)) {
+            return 0;
+        }
+        return wrappedContainer.extract(convertedSlot, resource, amount, transaction);
+    }
+
+    @Override
+    public int extract(ItemResource resource, int amount, TransactionContext transaction) {
+        int size = size();
+        int handled = 0;
+        for (int index = 0; index < size; index++) {
+            handled += extract(index, resource, amount - handled, transaction);
+            if (handled == amount) break;
+        }
+        return handled;
+    }
+
+    public void set(int index, ItemResource resource, int amount) {
```

Comment:
This method isn't used, can we get rid of this direct setting method so that people don't try to bypass transactions entirely?